x-pack/plugins/fleet/server/services/epm/packages/custom_integrations/validation/check_naming_collision.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { nodeBuilder } from '@kbn/es-query';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+
+import { installationStatuses } from '../../../../../../common/constants';
+import { auditLoggingService } from '../../../../audit_logging';
+import { PACKAGES_SAVED_OBJECT_TYPE, type Installation } from '../../../../../../common';
+
+export const checkForNamingCollision = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  integrationName: string
+) => {
+  const result = await savedObjectsClient.find<Installation>({
+    type: PACKAGES_SAVED_OBJECT_TYPE,
+    perPage: 1,
+    filter: nodeBuilder.and([
+      nodeBuilder.is(`${PACKAGES_SAVED_OBJECT_TYPE}.attributes.name`, integrationName),
+      // Installed packages
+      nodeBuilder.is(
+        `${PACKAGES_SAVED_OBJECT_TYPE}.attributes.install_status`,
+        installationStatuses.Installed
+      ),
+      // Custom packages
+      nodeBuilder.is(`${PACKAGES_SAVED_OBJECT_TYPE}.attributes.install_source`, 'custom'),
+    ]),
+  });
```

Comment:
As mentioned before, should we run this against all the install_source types instead of scoping it to `custom`? Also, what would happen if a user later installs a preset integration (let's say `apache`) from the Integrations page and we already have a custom integration with the same name? Would that override the custom integration?

Reply:
I agree to check for all `install_source` types, I think it's likely that installing a package from registry would override the custom integration, we should test it.
Could we use a specific name for custom integrations to avoid collisions? e.g. having a custom prefix or postfix

Reply:
I'd say we should also prevent the creation of custom integrations with same name of a registry package cause it will affect also the integrations page, here what happened when I retried to install `system` from the integrations page:
<img width="813" alt="Screenshot 2023-08-09 at 09 54 25" src="https://github.com/elastic/kibana/assets/34506779/7ad8aea9-08b5-443e-bfd0-3256b4799c98">

Reply:
Sure, I can change this to cover all sources.

@juliaElastic 
> Could we use a specific name for custom integrations to avoid collisions? e.g. having a custom prefix or postfix

Yeah, this would be easy to add. From the Fleet side would you have a preference for prefix or postfix?

Reply:
I don't have a strong preference.

Reply:
@juliaElastic @tonyghiani 

Hmm, so I've changed this to search across all installation source types, which is fine. However, I don't think the prefix (or postfix) idea for the name will work well. For example, the installed packages API utilises pagination, and is sorted by `name`, with the use of a custom prefix it means all the custom results get clumped together. For rendering we use the pretty `title`, but it still means these results are grouped next to each other and therefore the entries look incorrect (because they're ordered weird). Postfix will have similar issues also (when doing a name filter for example). 

I honestly think the collision check is enough (now we search across all sources), if the user gets the error they can attach their own prefix / postfix, or just alter the name.

I know Ruflin was also keen to stay away from the term "custom" 🤔

What do you think? I'd prefer to revert the prefix commits.

Reply:
I'm worried though if someone were to create a custom package first with the name `system`, then they will be blocked from installing the `system` package from registry. 
It would be good to at least prevent using the name of existing registry packages to name custom packages. Maybe we can add a validation for that?

Reply:
I see what you mean Kerry, this explains also why some entries in the selector were not correctly sorted in the discover log explorer.
I agree with Julia and I think our top priority rule for creating new custom integrations should be unicity in its name against any kind of existing integration, independently from the `install_source` or its `install_status`.

Not sure about it, but maybe removing the 
```
```

Reply:
I’ll enhance the validation case to block against naming collisions with packages that exist in the registry / are bundled.